optuna/samplers/_qmc.py

```diff
@@ -0,0 +1,329 @@
+from collections import OrderedDict
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+import numpy as np
+
+# We want to lazy import scipy.stats.qmc as it is slow to import.
+import scipy  # NOQA
+
+import optuna
+from optuna import distributions
+from optuna import logging
+from optuna._experimental import experimental
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = logging.get_logger(__name__)
+
+_SUGGESTED_STATES = (TrialState.COMPLETE, TrialState.PRUNED)
+
+_NUMERICAL_DISTRIBUTIONS = (
+    distributions.UniformDistribution,
+    distributions.LogUniformDistribution,
+    distributions.DiscreteUniformDistribution,
+    distributions.IntUniformDistribution,
+    distributions.IntLogUniformDistribution,
+)
+
+
+@experimental("3.0.0")
+class QMCSampler(BaseSampler):
+    """A Quasi Monte Carlo Sampler that generates low-discrepancy sequences.
+
+    Quasi Monte Carlo (QMC) sequences are designed to have lower discrepancies than
+    standard random seqeunces. They are known to perform better than the standard
+    randam sequences in hyperparameter optimization.
+
+    For further information about the use of QMC sequences for hyperparameter optimization,
+    please refer to the following paper:
+
+    - Bergstra, James, and Yoshua Bengio. Random search for hyper-parameter optimization.
+      Journal of machine learning research 13.2, 2012.
+      <https://jmlr.org/papers/v13/bergstra12a.html>`_
+
+    We use the QMC implementations in Scipy. For the details of the QMC algorithm,
+    see the Scipy API references on `scipy.stats.qmc
+    <https://scipy.github.io/devdocs/stats.qmc.html>`_.
```

Comment:
```suggestion
    <https://scipy.github.io/devdocs/reference/stats.qmc.html>`_.
```

The link was seemingly changed.